lib/experimental/block-editor-settings.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Adds settings to the block editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds styles and __experimentalFeatures to the block editor settings.
+ *
+ * @param array $settings Existing block editor settings.
+ *
+ * @return array New block editor settings.
+ */
+function gutenberg_get_block_editor_settings_experimental( $settings ) {
+	$is_block_theme = wp_is_block_theme();
+
+	if ( ! $is_block_theme ) {
+		return $settings;
+	}
+
+	global $post_id;
+
+	$template_slug = get_page_template_slug( $post_id );
+
+	if ( ! $template_slug ) {
+		$post_slug      = 'singular';
+		$page_slug      = 'singular';
+		$template_types = get_block_templates();
+
+		foreach ( $template_types as $template_type ) {
+			if ( 'page' === $template_type->slug ) {
+				$page_slug = 'page';
+			}
+			if ( 'single' === $template_type->slug ) {
+				$post_slug = 'single';
+			}
+		}
+
+		$what_post_type = get_post_type( $post_id );
+		switch ( $what_post_type ) {
+			case 'post':
+				$template_slug = $post_slug;
+				break;
+			case 'page':
+				$template_slug = $page_slug;
+				break;
+		}
+	}
+
+	$current_template = gutenberg_get_block_templates( array( 'slug__in' => array( $template_slug ) ) );
```

Comment:
Is there any possibility that `$template_slug` will be `null` at this stage? E.g. what happens if someone is editing a custom post type (not post or page)? I don't imagine anything bad would happen, but just a thought in case there's any benefit in bailing early 🤔

Reply:
Just checked this and it would be null for a custom post. I'm changing the switch above so that basically anything that's not a page will get the default post template assigned, that should cover everything!

Reply:
Done ✔️